projects/ngx-operators/src/lib/upload.ts

```diff
@@ -1,30 +1,33 @@
-import { HttpEvent } from "@angular/common/http";
+import { HttpEvent, HttpResponse } from "@angular/common/http";
 import { Observable } from "rxjs";
 import { distinctUntilChanged, scan } from "rxjs/operators";
 import { isHttpProgressEvent, isHttpResponse } from "./http";
 
-export interface Upload {
+export interface Upload<T> {
   progress: number;
   state: "PENDING" | "IN_PROGRESS" | "DONE";
+  body: T | null;
 }
 
-export function upload(): (
-  source: Observable<HttpEvent<unknown>>
-) => Observable<Upload> {
-  const initialState: Upload = { state: "PENDING", progress: 0 };
-  const reduceState = (state: Upload, event: HttpEvent<unknown>): Upload => {
+export function upload<T>(): (
```

Comment:
How about we default the generic parameter to `unknown`? That way this wouldn't be a breaking change for users of the library

```suggestion
export function upload<T = unknown>(): (
```

Reply:
This makes sense. Fixed